spyder/utils/ipython/kernelspec.py

```diff
@@ -43,6 +44,9 @@ def argv(self):
             # to the kernel sys.path
             os.environ.pop('VIRTUAL_ENV', None)
             pyexec = CONF.get('main_interpreter', 'executable')
+            if not is_python_interpreter(pyexec):
+                pyexec = get_python_executable()
+                CONF.reset_to_defaults(section='main_interpreter')
```

Comment:
We don't need to reset the entire section. What we need to do is

1. Set `'main_interpreter', 'executable'` to be empty.
2. Make `'main_interpreter', 'default'` True
3.  Make `'main_interpreter', 'custom'` False.